sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -647,6 +650,21 @@ RTDeviceBinaryImage &ProgramManager::getDeviceImage(OSModuleHandle M,
 
   Ctx->getPlugin().call<PiApiKind::piextDeviceSelectBinary>(
       getFirstDevice(Ctx), RawImgs.data(), (cl_uint)RawImgs.size(), &ImgInd);
+
+  if (CheckThatCompiled) {
+    // If the image is already compiled with AOT, throw an exception.
```

Comment:
I guess we don't need to throw an exception if there is a SPIR-V source. Right?

Reply:
No, we should throw an exception because we do not support recompilation.

Reply:
I don't agree here. Compiling SPIR-V is not "recompilation".
Even we have pre-compiled binary for some device, it should be a valid action to compile/build at runtime the program for the same device.
But I'm okay if we enable this use case in a separate PR.